sklearn/decomposition/incremental_pca.py

```diff
@@ -210,11 +210,19 @@ def partial_fit(self, X, y=None, check_input=True):
             self.components_ = None
 
         if self.n_components is None:
-            self.n_components_ = n_features
+            if self.components_ is None:
+                self.n_components_ = min(n_samples, n_features)
+            else:
+                self.n_components_ = self.components_.shape[0]
         elif not 1 <= self.n_components <= n_features:
             raise ValueError("n_components=%r invalid for n_features=%d, need "
                              "more rows than columns for IncrementalPCA "
                              "processing" % (self.n_components, n_features))
+        elif not self.n_components <= n_samples:
+            raise ValueError("n_components=%r must be less or equal to "
+                             "the batch number of samples %d. You can change "
```

Comment:
Is that so. I added that to avoid putting the emphasis on either of the parameters. But you are right that it should be easy to see even for a beginner.